reverie/backend_server/persona/prompt_template/gpt_structure.py

```diff
@@ -191,6 +191,40 @@ def ChatGPT_request(prompt):
     traceback.print_exc()
     return "LLM ERROR"
 
+def ChatGPT_structured_request(prompt, response_format):
+  """
+  Given a prompt and a dictionary of GPT parameters, make a request to OpenAI
+  server and returns the response. 
+  ARGS:
+    prompt: a str prompt
+    gpt_parameter: a python dictionary with the keys indicating the names of  
+                   the parameter and the values indicating the parameter 
+                   values.   
+  RETURNS: 
+    a str of GPT-3's response. 
+  """
+  # temp_sleep()
+  print("--- ChatGPT_request() ---")
+  print("Prompt:", prompt)
+
+  try: 
+    completion = client.beta.chat.completions.parse(
+      model=openai_config["model"],
+      response_format=response_format,
+      messages=[{"role": "user", "content": prompt}]
+    )
+    content = completion.choices[0].message.content
+    print("Response content:", content)
+    cost_logger.update_cost(
+      completion, input_cost=openai_config["model-costs"]["input"], output_cost=openai_config["model-costs"]["output"]
+    )
+    return content
```

Comment:
Here you'll want to work with and return `message.parsed` like in the other low-level structured outputs function, the one that I made. Then you'll need to bubble that up through the higher-level functions so they work with an object rather than a string